relay-server/src/actors/project.rs

```diff
@@ -421,6 +424,26 @@ impl Project {
         }
     }
 
+    /// Returns the [`ReservoirCounters`] for the project.
+    pub fn reservoir_counters(&self) -> ReservoirCounters {
+        self.reservoir_counters.clone()
+    }
+
+    /// If a reservoir rule is no longer in the sampling config, we can assume it's no longer needed.
+    pub fn remove_expired_reservoir_rules(&self) {
+        let Some(config) = self
+            .state
+            .as_ref()
+            .and_then(|state| state.config.dynamic_sampling.as_ref())
+        else {
+            return;
+        };
+
+        if let Ok(mut guard) = self.reservoir_counters.try_lock() {
```

Comment:
Using `try_lock` comes with the downside of skipping cleanup in case the rules are being used right now. It is an unlikely case, so it's a fair assumption, but please leave a code comment of why this is so crucial to prevent that someone changes this into a full `lock()` in the future. 

The only way to lock this here is using a tokio Mutex, but before we do that we should explore other approaches.